src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -157,7 +153,8 @@ public abstract class AbstractBuilding extends AbstractBuildingContainer
     /**
      * Set of building modules this building has.
      */
-    protected List<IBuildingModule> modules = new ArrayList<>();
+    protected List<IBuildingModule>                  modules    = new ArrayList<>();
+    protected Int2ObjectOpenHashMap<IBuildingModule> modulesMap = new Int2ObjectOpenHashMap<>();
```

Comment:
arraylist is better for when we iterate all of them, when e.g. searching or using class lookups. Since its a very small memory overhead I thought it doesnt hurt to keep both for better access speed